[NTH-59] - Signup page

ahmedmurtaja opened the pull request:
- make a signup page for patients & therapist
- handle rendering additional fields
- handle scroll behavior 
- fix styling problem and image resizing

ahmedmurtaja committed:
feat: make sign up page

ahmedmurtaja committed:
feat: add scroll behaviour to the sign up page

ahmedmurtaja committed:
Refactor : refactor code of rendering additional elements

LinaYahya commented on src/pages/signup/index.tsx:
```diff
@@ -1,5 +1,225 @@
+};
+
+const Signup = () => {
+  const [userType, setUserType] = useState('');
```
let us have a default value as a user

ahmedmurtaja replied:
done

LinaYahya commented on src/pages/signup/index.tsx:
```diff
@@ -1,5 +1,225 @@
+
+const Signup = () => {
+  const [userType, setUserType] = useState('');
+  const [showAdditionalFields, setShowAdditionalFields] = useState(false);
```
No need for this as a state, actually we can render additional inputs based on user type

ahmedmurtaja replied:
done

LinaYahya commented on src/pages/signup/index.tsx:
```diff
@@ -1,5 +1,225 @@
+                style={{ margin: '10px' }}
+                value="patient"
+                control={<Radio />}
+                label="patient"
```
```suggestion
                label="user"
```
And I'd prefer to have user radio btn before therapist

ahmedmurtaja replied:
done

LinaYahya commented on src/pages/signup/index.tsx:
```diff
@@ -1,5 +1,225 @@
+              sx={textFieldStyle}
+              id="username"
+              name="username"
+              label="username"
```
```suggestion
              label="Full Name"
```

ahmedmurtaja replied:
done

LinaYahya commented on src/pages/signup/index.tsx:
```diff
@@ -1,5 +1,225 @@
+              sx={textFieldStyle}
+              id="confirmPassword"
+              name="confirmPassword"
+              label="confirmPassword"
```
```suggestion
              label="Confirm Password"
```

ahmedmurtaja replied:
done

LinaYahya commented on src/pages/signup/schema.ts:
```diff
@@ -0,0 +1,34 @@
+    .required('Major is required'),
+  hourlyRate: yup
+    .number()
+    .required('Hourly Rate is required'),
```
can we have a check for CV and profile image, OR Just to handle it later after take decision of how to upload it?
@ahmedmurtaja

ahmedmurtaja replied:
I think we can handle it later in the S3 task, but for now, I think of using [React-Toastify](https://fkhadra.github.io/react-toastify/) for handling successful or failed file upload

LinaYahya commented on src/pages/signup/index.tsx:
```diff
@@ -1,5 +1,225 @@
+            id="major"
+            name="major"
+            label="major"
+            type="major"
```
```suggestion
            type="major"
```
🤔

LinaYahya commented on src/pages/signup/index.tsx:
```diff
@@ -1,5 +1,225 @@
+            id="hourlyRate"
+            name="hourlyRate"
+            label="hourlyRate"
+            type="hourlyRate"
```
input type is not associated with it's name, Such common input types (email, name, number, password)

LinaYahya commented on src/pages/signup/index.tsx:
```diff
@@ -1,5 +1,225 @@
+            <TextField
+              margin="normal"
+              required
+              autoFocus
```
```suggestion
```

LinaYahya commented on src/pages/signup/index.tsx:
```diff
@@ -1,5 +1,225 @@
+              id="email"
+              name="email"
+              label="Email"
+              autoFocus
```
```suggestion
```
AutoFoxus gonna led to focus last input which I find it a bit annoying, SO I'd prefer to delete this attribute for all inputs

LinaYahya reviewed (changes requested):
Good Job Ahmed, A small notes to take in consideration, 
As well as we need to have a show/hide password functionality 
[Input Adornments](https://mui.com/material-ui/react-text-field/#input-adornments), and I guess changes u made affect login background image, can u check?

ahmedmurtaja committed:
fix: fix autoFoucus,Type errors

ahmedmurtaja committed:
fix: fix user radio

ahmedmurtaja committed:
chore:refactor file upload method

ahmedmurtaja committed:
fix: fix login image

ahmedmurtaja commented:
> Good Job Ahmed, A small note to take into consideration, As well as we need to have a show/hide password functionality [Input Adornments](https://mui.com/material-ui/react-text-field/#input-adornments), and I guess the changes u made affect login background image, can u check?

Sorry, I changed the login style by mistake (I fixed it now).

I will make a task to fix the password field on Login Page also after fixing it in this branch

ahmedmurtaja committed:
feat: add React-Toastify alert

ahmedmurtaja committed:
feat: add show/hide password functionality

ahmedmurtaja committed:
feat: add InputAdornment dolar sign to the start of HourlyRate Input

ahmedmurtaja committed:
Refactor: remove ract toastify package and use MUI ALERT

ahmedmurtaja committed:
Refactor: use notistack

ahmedmurtaja committed:
Refactor: use snackbar in main

ahmedmurtaja committed:
fix: fix image resizing problem
